src/sock.c

```diff
@@ -475,6 +476,45 @@ static void mg_iotest(struct mg_mgr *mgr, int ms) {
     FreeRTOS_FD_CLR(c->fd, mgr->ss,
                     eSELECT_READ | eSELECT_EXCEPT | eSELECT_WRITE);
   }
+#else
+#ifdef MG_ENABLE_POLL
+  struct mg_connection *c;
+  size_t n = 0;
+  for (c = mgr->conns; c != NULL; c = c->next) n++;
+  struct pollfd *fds = malloc(n * sizeof(struct pollfd));
+  int rc;
+  size_t i = 0;
+
+  for (c = mgr->conns; c != NULL; c = c->next) {
+      if (c->is_closing || c->is_resolving || FD(c) == INVALID_SOCKET) {
+          i++;
+          continue;
+      }
+      fds[i].fd = FD(c);
+      fds[i].events = 0;
+      fds[i].events |= POLLIN;
+      if (c->is_connecting || (c->send.len > 0 && c->is_tls_hs == 0)) {
+          fds[i].events |= POLLOUT;
+      }
+      i++;
+  }
+
+  if ((rc = poll(fds, n, ms)) < 0) {
+      MG_ERROR(("poll: %d %d", rc, MG_SOCK_ERRNO));
+  }
+
+  i = 0;
+  for (c = mgr->conns; c != NULL; c = c->next) {
```

Comment:
If you stick i++ into the for(), then you don't need to care about increments in the loop body
for (c = mgr->conns; c != NULL; c = c->next, i++)